### PR TITLE
Optimise Github API calls

### LINKED ISSUE #348: Reauthorization request encountered while using WATcher
After using WATcher for just a few minutes (loaded less than 5 repos), I encountered this:
image

### PR DESCRIPTION
**Pull request by nknguyenhc**
### Summary:

Fixes #348

#### Type of change:

- ✨ New Feature/ Enhancement

### Changes Made:

* Remove redundant API calls
* Group duplicate API calls

### Proposed Commit Message:

```
Optimise Github API calls

Previously, multiple Github API calls causes Github
to ask users for reauthorisation, or in rare cases,
rate limit is exceeded.

We reduce the number of API calls to Github
to reduce such issues.
```

<details><summary>
<h3>Checklist:</h3>
</summary>

- [ ] I have tested my changes thoroughly.
- [ ] I have created tests for any new code files created in this PR or provided a link to a issue/PR that addresses this.
- [ ] I have added or modified code comments to improve code readability where necessary.
- [ ] I have updated the project's documentation as necessary.

</details>

### CODE REVIEW

**Comment by nknguyenhc** [comment]
Currently, entering CATcher-org/WATcher takes around 20 API calls.

image

Subsequently switching to `AY2324S1-CS2103T-T08-1/tp`, `AY2324S2-CS2103T-T10-2/tp`, `AY2324S2-CS2103T-T10-1/tp`, `AY2324S2-CS2103T-T10-3/tp`, `AY2324S2-CS2103T-T09-3/tp` takes ~200 API calls.

image

Reloading WATcher at this point does not raise a reauthorisation from Github.

Does not seem to encounter any more prompt from Github for reauthorisation (due to high number of API calls). @CATcher-org/2324s2 do test out and let me know.

**Review by NereusWB922 — APPROVED** [review]
LGTM!

File: src/app/shared/filter-bar/label-filter-bar/label-filter-bar.component.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export class LabelFilterBarComponent implements OnInit, AfterViewInit, OnDestroy
         this.selectedLabelNames = new Set<string>(this.filtersService.filter$.value.labels);
         this.deselectedLabelNames = this.filtersService.filter$.value.deselectedLabels;
         this.hiddenLabelNames = this.filtersService.filter$.value.hiddenLabels;
+        this.loaded = true;
```
Comment (Arif-Khalid) [inline src/app/shared/filter-bar/label-filter-bar/label-filter-bar.component.ts:43]: Excellent refactor into an existing subscription

File: src/app/core/services/github.service.ts [hunk 2]
```diff
@@ -115,31 +115,25 @@ export class GithubService {
...
+            { owner: ORG_NAME, name: REPO, filter: graphqlFilter },
+            (result) => result.data.repository.issues.edges,
+            GithubGraphqlIssueOrPr
+          ),
+          this.fetchGraphqlList<FetchPullRequestsQuery, GithubGraphqlIssueOrPr>(
+            FetchPullRequests,
+            { owner: ORG_NAME, name: REPO },
+            (result) => result.data.repository.pullRequests.edges,
+            GithubGraphqlIssueOrPr
+          )
```
Comment (Arif-Khalid) [inline src/app/core/services/github.service.ts:133]: Much cleaner implementation!

**Review by Arif-Khalid — APPROVED** [review]
A great PR which shows attention to detail and care for code quality. LGTM!

**Pull request merged**